zebra/zapi_msg.c

```diff
@@ -2962,6 +2962,41 @@ static inline void zread_iptable(ZAPI_HANDLER_ARGS)
 	return;
 }
 
+static inline void zread_neigh_discover(ZAPI_HANDLER_ARGS)
+{
+	struct stream *s;
+	ifindex_t ifindex;
+	struct interface *ifp;
+	struct prefix p;
+	struct ipaddr ip;
+
+	s = msg;
+
+	STREAM_GETL(s, ifindex);
+
+	ifp = if_lookup_by_index_per_ns(zvrf->zns, ifindex);
+	if (!ifp) {
+		zlog_debug("Failed to lookup ifindex: %u", ifindex);
+		return;
+	}
+
+	STREAM_GETC(s, p.family);
+	STREAM_GETC(s, p.prefixlen);
+	STREAM_GET(&p.u.prefix, s, prefix_blen(&p));
+
+	if (p.family == AF_INET)
+		SET_IPADDR_V4(&ip);
+	else
+		SET_IPADDR_V6(&ip);
+
+	memcpy(&(ip.ip.addr), &(p.u.prefix), prefix_blen(&p));
```

Comment:
nit: redundant parentheses

Reply:
I found this easier to read, but now I see that this only adds inconsistencies, so I removed them. Thanks!